api/server/cluster.go

```diff
@@ -40,6 +41,7 @@ func (c *clusterApi) Routes() []*Route {
 		{verb: "PUT", path: clusterPath("/shutdown/{id}", cluster.APIVersion), fn: c.shutdown},
 		{verb: "GET", path: clusterPath("/alerts/{resource}", cluster.APIVersion), fn: c.enumerateAlerts},
 		{verb: "PUT", path: clusterPath("/alerts/{resource}/{id}", cluster.APIVersion), fn: c.clearAlert},
+		{verb: "POST", path: clusterPath("/setclustersecretkey", cluster.APIVersion), fn: c.setClusterSecretKey},
```

Comment:
Should this go into api/server/secrets.go , where secret.go will implement restserver interface from api/server/server.go